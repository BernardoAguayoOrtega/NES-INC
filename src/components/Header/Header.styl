header
	width: 100%
	height: 60px
	background-color:#36404e
header nav
		display: flex
		justify-content: space-between

header nav .nav-right-section
	height: auto
	display: flex
	list-style: none
	justify-content: flex-end
	align-items:center

nav .nav-right-section a
	margin-right: 10px
	text-decoration:none
	color:white
