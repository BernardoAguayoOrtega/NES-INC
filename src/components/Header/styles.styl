//variables
@import "../../stylus/variables.styl";

.header
	width: 100%
	height: 100px
	display: flex
	justify-content: center
	background-color:color_primary
	position: sticky
	top:0
	z-index: 11

	.nav
		display: flex
		justify-content: space-between
		align-items: center
		max-width: 1200px
		&__logo
			margin: 0
			height: 100px

		&__right
			height: auto
			display: flex
			list-style: none
			justify-content: flex-end
			align-items:center

			span,a
				margin-right: 10px
				text-decoration:none
				color:white

			&--menu
				margin-bottom: 5px

			.search i
				transform: scaleX(-1)

			#searchform
				display: none
				margin-right: 10px

				input
					margin: 0 0 10px 0
					padding: 2px 2px
					background: #fff
					border: 1px solid gray
					color: #303030
					height: 24px
					font-size: 16px
					padding-top: 0px
					padding-left: 10px
					background-size: 20px 300px

@media screen and (max-width: 1200px)
	.nav
		width: 100%
@media screen and (min-width: 1200px)
	.nav
		width: 1200px
