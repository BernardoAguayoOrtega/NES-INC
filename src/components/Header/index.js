// import react
import React from 'react';

import '../Header/Header.styl'
import Logo from '../../assets/Logo' 

//create and export app component
const Header = () =>  (
  <header>
		<nav>
			<a href="#" className="icon_logo">
				<Logo />
			</a>
			<ul className="nav-right-section">
				<li>
					<a href="#">Search</a>
				</li>
				<li className="menu_icon">
					<a href="#">Menu</a>
				</li>
			</ul>
		</nav>
	</header>
);


export default Header;
