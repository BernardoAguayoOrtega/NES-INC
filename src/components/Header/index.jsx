// import react
import React, { useContext } from 'react';
// import link from react router dom
import { Link } from 'react-router-dom';
// import Logo
import Logo from '../../assets/Logo/index.js';
// import menu component
import Menu from '../Menu/index.jsx';
// import search component
import Search from '../Search/index.jsx';
// import context
import { Context } from '../../utils/Context/index.jsx';
// import header styles
import './styles.styl';

//create and export app component
const Header = () => {
  const { setAction } = useContext(Context);

  return (
    <header className="header">
      <nav className="nav">
        <Link to="/" onClick={() => setAction({ option: 0 })}>
          <Logo className="nav__logo" />
        </Link>
        <ul className="nav__right">
          <li className="nav__right--search">
            <Search />
          </li>
          <li className="nav__right--menu">
            <span id="arrow">
              <Menu />
            </span>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
