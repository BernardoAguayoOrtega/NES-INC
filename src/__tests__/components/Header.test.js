// import react
import React from 'react';
// import shallow
import { shallow, mount } from 'enzyme';
// import router
import { HashRouter as Router } from 'react-router-dom';
// import context provider
import { ContextProvider } from '../../utils/Context/index.jsx';
// import toJson
import toJson from 'enzyme-to-json';
// import app component
import Header from '../../components/Header/index.jsx';

describe('<Header /> shallow', () => {
  const wrapper = shallow(
    <ContextProvider>
      <Router>
        <Header />
      </Router>
    </ContextProvider>,
  );

  it('is the Header rendering?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('is there a header tag?', () => {
    expect(wrapper.find('header').exists()).toBe(false);
  });

  it('header has an only child?', () => {
    expect(wrapper.find('header').children().length).toBe(0);
  });

  it('matches the snap shot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('<Header /> mount', () => {
  const wrapper = mount(
    <ContextProvider>
      <Router>
        <Header />
      </Router>
    </ContextProvider>,
  );

  it('is the Header rendering?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('is there a header tag?', () => {
    expect(wrapper.find('header').exists()).toBe(true);
  });

  it('header has an only child?', () => {
    expect(wrapper.find('header').children().length).toBe(1);
  });

  it('header has a link component?', () => {
    expect(wrapper.find('Link').exists()).toBe(true);
  });

  it('does the header link component has the correct route?', () => {
    expect(wrapper.find('Link').first().prop('to')).toBe('/');
  });
});
