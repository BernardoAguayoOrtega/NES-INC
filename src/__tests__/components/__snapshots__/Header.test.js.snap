// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> shallow matches the snap shot 1`] = `
<ContextProvider
  value={
    Object {
      "news": Object {
        "data": Array [
          Object {
            "images": "",
            "title": "",
          },
          Object {
            "images": "",
            "title": "",
          },
          Object {
            "images": "",
            "title": "",
          },
          Object {
            "images": "",
            "title": "",
          },
          Object {
            "images": "",
            "title": "",
          },
          Object {
            "images": "",
            "title": "",
          },
        ],
        "loading": true,
      },
      "setAction": [Function],
    }
  }
>
  <HashRouter>
    <Header />
  </HashRouter>
</ContextProvider>
`;
